src/cfml/system/services/ServerService.cfc

```diff
@@ -1584,11 +1583,17 @@ component accessors="true" singleton {
 			menuItem[ 'path' ] = fileSystemUtil.resolvePath( menuItem.path, relativePath );
 		}
 
+		//checking if the action is the legacy "restartserver", it needs to be translated to a box type action
+		if( menuItem.keyExists( 'action' ) && listFindNoCase('restartserver',menuItem.action)){
+			menuItem[ 'action' ] = "box server restart " & '"#name#"';
+		}
+
```

Comment:
@mgmathus This is not the approach we discussed.  I told you to modify the menu item on line 1040 to be of type `runasync`.  None of the code in this function needed touched except the expansion of the box binary.  The `restartserver` menu item action doesn't exist any longer.